### PR TITLE
Rate limit the requests. Fix #49.

### LINKED ISSUE #49: Feature to control messages per second
When I send many requests, I get this error:

> Max rate of messages per second has been exceeded:max=50 rec=108

### PR DESCRIPTION
**Pull request by claude2**
Limiting requests to 40 per second. 45 still generated warnings that we were approaching the max message rate.

PS Please [squash](https://help.github.com/articles/about-pull-request-merge-squashing/) for the merge, thanks

--
Related Issue: #49

### CODE REVIEW

File: lib/outgoing.js [hunk 1]
```diff
@@ -14,10 +16,10 @@ function Outgoing(controller) {
   this._controller = controller;
 }
 
-Outgoing.prototype._send = function () {
+Outgoing.prototype._send = rateLimit(40, 1000, function () {
```
Comment (pilwon) [inline lib/outgoing.js]: @claude2 Could we write these magic numbers in `lib/constants.js`?

**Comment by pilwon** [comment]
@claude2 Could you share your insights on why you chose `function-rate-limit` over other alternatives (ex: `lodash.throttle`)?

**Comment by claude2** [comment]
Debounce and [throttle](https://lodash.com/docs#throttle) drop events. `function-rate-limit` queues them up, is a simple package that works, was updated recently, several [other packages depend on it](https://www.npmjs.com/package/function-rate-limit), and it has 8k downloads per month. [limiter](https://www.npmjs.com/package/limiter) is more popular, but has a strange API.

**Comment by pilwon** [comment]
@claude2 Thanks for your PR and also for sharing your insights.

**Pull request merged**